### PR TITLE
FEAT: Add non-nullable info to schema output

### LINKED ISSUE #2115: Schema output doesn't show if the field is non-nullable
The current schema doesn't show if the field is non-nullable

```python
>>> ibis.schema([
...     ('foo', 'int64'),
...     ('bar', ibis.expr.datatypes.int64(nullable=False)),
...     ('baz', 'boolean*')
... ])

ibis.Schema {  
  foo  int64
  bar  int64
  baz  boolean
}

```

*Desired behavior*:

The output should show if the type is non-nullable, eg:

```
ibis.Schema {  
  foo  int64
  bar  int64[non-nullable]
  baz  boolean
}
```

### PR DESCRIPTION
**Pull request by xmnlab**
Resolves #2115

### CODE REVIEW

**Comment by xmnlab** [comment]
It seems the error

```
   ValueError: Incompatible component merge:
      - 'cp38'
      - '*_cp38'
```

was already fixed.  maybe in 1 hour or more  it would be available.

**Comment by xmnlab** [comment]
https://github.com/conda-forge/python-feedstock/pull/325

**Comment by xmnlab** [comment]
@jreback this PR is ready for review. thanks!

File: ibis/expr/datatypes.py [hunk 1]
```diff
@@ -83,7 +83,9 @@ def __repr__(self) -> str:
         )
 
     def __str__(self) -> str:
-        return self.name.lower()
+        return '{}{}'.format(
```
Comment (jreback) [inline ibis/expr/datatypes.py]: instead of this you can simply define
```
__str__ = __repr__
```
as repr already does this.

Comment (xmnlab) [reply ibis/expr/datatypes.py]: actually they are not the same.

with the current approach it prints:

```
ibis.Schema {  
  foo  int64
  bar  int64[non-nullable]
  baz  boolean
}
```

using __str__ = __repr__, it prints:

```
ibis.Schema {  
  foo  Int64(nullable=True)
  bar  Int64(nullable=False)
  baz  Boolean(nullable=True)
}
```

**Comment by xmnlab** [comment]
release note added and ci green.

**Comment by jreback** [comment]
lgtm. pls resolve conflicts and ping on green.

**Comment by xmnlab** [comment]
thanks for the review @jreback ! ci is green :)

**Comment by jreback** [comment]
thanks @xmnlab

**Pull request merged**